docs_src/design/adr/devops/0010-Release-Artifacts.md

```diff
@@ -0,0 +1,84 @@
+
+#### Nexus Retention Policy
+
+##### docker.snapshots
+Retention Policy: 90 days since last download
+
+Contains: Docker images that are not expected to be released. This will contain EdgeX that images are built with each merge and should be available to test things against a specific commit. In addition to this we also build images to optimize the builds in the CI infrastructure.
+
+Docker Tags Used: Git commit, Version-Git commit, Version (ci-build-images only), Latest (ci-build-images-only)
+
+##### docker.staging
+Retention Policy: 180 days since last download
+
+Contains: Docker images built for potential release and testing purposes.
+
+Docker Tags Used: Version (ie: v1.x), Release Branch (master, fuji, etc), Latest
+
+##### docker.release
+Retention Policy: No automatic removal. Requires TSC approval to remove images from this repository.
+
+Contains: Officially released docker images for EdgeX. 
+
+Docker Tags Used:•Version (ie: v1.x), Latest 
+
+[Nexus Cleanup Policies Reference](https://help.sonatype.com/repomanager3/cleanup-policies)
+
+### Docker Compose Files
+
+*Tied to Code Release?* Yes
+
+Docker compose files are released alongside the docker images for every release of EdgeX Foundry. During development the community maintains a compose file named `nightly-build`. This compose file is meant to be a beta version of EdgeX. At the time of release the community makes a named release compose file. (ie: `geneva`)
+
+### Github Page: EdgeX Docs
+
+*Tied to Code Release?* No
+
+EdgeX Foundry releases a set of documentation for our project at http://docs.edgexfoundry.org. This page is a github page that is managed by the `edgex-docs` Github repository. As a community we make our best effort to keep these docs up to date. On this page we are also versioning the docs with the semantic versions of the named releases. As a community we try to version our documentation site shortly after the official release date but documentation changes are addressed as we find them throughout the release cycle.
+
+### GitHub Tags
+
+*Tied to Code Release?* Yes, for the final semantic version
+
+Github tags are used to track the releases of EdgeX Foundry. During development the tags are incremented automatically using a development suffix (ie: `v1.1.1-dev.1` -> `v1.1.1-dev.2`). At the time of release we release a tag with the final semantic version (ie: `v1.1.1`).
+
+### Snaps
+
+*Tied to Code Release?* Yes
+
+Snaps are released for every named release of EdgeX Foundry. During development the community stages snaps to the `latest/edge` channel of the [Snapcraft Store](https://snapcraft.io). At the time of code freeze we will promote the snaps from `latest/edge` to the `latest/beta` channels as beta release. At the time of release we will promote the snaps from `latest/beta` to `latest/stable`. In addition we also create a named release track for the release. (ie: `geneva/stable`).
```

Comment:
also what was the decision re: versioning and tagging the various snap releases so that the version field in the snap matches the release tag? I.e. so we don't end up with `version: v1.1.1-dev.2` in the snap on latest/stable ? 

I remember suggesting that the release job download the built snap artifacts on the latest/beta channel, unpack and change the version number, then re-upload and release to the latest/stable channel.

Reply:
During the dot release we had more issues with releasing the dev versions not only in snaps but in the docker images as well. We found that the version api endpoint for the services was returning the dev versions in both. To fix this in the dot release I did a manual rebuild of everything. 

For Hanoi I'm going to make an update to the automation ADR as well. Essentially at code freeze we will need to rebuild the release candidates with the proper `version (ie: v1.1.x)`. For snaps this means at beginning of code freeze we will push that candidate to the `beta/candidate` channels for integration testing. At the end of the code freeze we should be able to promote that candidate to the `stable` channel. 

For docker images it's similar because we are doing an image promotion from `docker.staging` to `docker.release+Dockerhub` at release time. So during code freeze we will have to build those images and push them to staging to get ready for the release.

This means that during code freeze we will have to rebuild the artifacts each time we fix found issues too.

Reply:
sounds good to me, thanks